### PR TITLE
sherpa: remove paths to compiler wrappers

### PR DESCRIPTION
**Pull request by jmcarcell**
These are the fixes to a couple of issues found in https://github.com/key4hep/key4hep-spack/issues/575 after installation.
- One of the issues is that paths to the spack compiler wrappers are hardcoded and then used later in makelibs to try to compile a process. This is fixed by simply removing those paths, and letting the current compiler be used.
- The other issue is that the libtools that was installed was different from the system one, and the system one is the one being picked up by autoreconf unless we add a path to aclocal.m4. This causes problems later because a version check is done and the process that is produced after running Sherpa can't be compiled.

The file has to be changed in the install phase because only then those parts are available, I also changed dependencies like libtool to also be a runtime dependency since they are needed at runtime.

### CODE REVIEW

**Comment by vvolkl** [comment]
LGTM, but you might want to add a comment to the autotools dependencies - It's unusual that they're a runtime dependency, and someone just looking at the recipe might try to "fix" this.

**Comment by jmcarcell** [comment]
Comment added.

**Review by vvolkl — DISMISSED** [review]
Great, thanks! Waiting with the merge to give @wdconinc a chance to review.

File: var/spack/repos/builtin/packages/sherpa/package.py [hunk 1]
```diff
@@ -159,10 +159,12 @@ class Sherpa(AutotoolsPackage):
...
-    depends_on("autoconf", type="build")
-    depends_on("automake", type="build")
-    depends_on("libtool", type="build")
-    depends_on("m4", type="build")
+    # autotools dependencies are needed at runtime to compile processes
+    # at least as long as sherpa is an autotools package
+    depends_on("autoconf")
+    depends_on("automake")
+    depends_on("libtool")
+    depends_on("m4")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/sherpa/package.py:167]: From the comments here and on the key4hep issue, it should only be libtool that's needed at runtime, right?

File: var/spack/repos/builtin/packages/sherpa/package.py [hunk 2]
```diff
@@ -261,3 +263,17 @@ def flag_handler(self, name, flags):
                     flags.append("-m64")
 
         return (None, None, flags)
+
+    # This may not be needed when this package is changed to be a CMake package
+    # since it's specific to makelibs
+    def install(self, spec, prefix):
+        # Remove the path to the spack compiler wrappers
+        # We don't want to use the paths to the wrappers later on
+        filter_file(r".*/spack/lib/spack.*", "", "AMEGIC++/Main/makelibs")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/sherpa/package.py]: The first `spack` in the path is where you may have installed spack but it is not a guarantee. I have (work)trees with e.g. `spack-v0.21.2` instead. It should be SPACK_ROOT.

Comment (wdconinc) [reply var/spack/repos/builtin/packages/sherpa/package.py]: In fact, you may want to use the spack function for this, e.g. https://github.com/spack/spack/blob/07fb83b493d522ef5369e4db565841674b4512c1/var/spack/repos/builtin/packages/rivet/package.py#L73

File: var/spack/repos/builtin/packages/sherpa/package.py (hunk 1 above)
Comment (jmcarcell) [reply var/spack/repos/builtin/packages/sherpa/package.py:167]: What is done at runtime is to do a full configure - compilation cycle over generated processes so I think all the tools are needed.

File: var/spack/repos/builtin/packages/sherpa/package.py (hunk 2 above)
Comment (jmcarcell) [reply var/spack/repos/builtin/packages/sherpa/package.py]: I didn't know about `filter_compiler_wrappers`; I changed the first `filter_file` to that but I haven't tested it.

**Comment by jmcarcell** [comment]
@spackbot style

**Review by wdconinc — DISMISSED** [review]
Looks fine to me now, if this does indeed to the right thing with the `filter_compiler_wrappers` call.

**Comment by jmcarcell** [comment]
Now it should be good to go; the `filter_compiler_wrappers` takes a path relative to the installation folder and not to the build folder and in this case they happen to be different, so I updated the correct path.

**Review by wdconinc — APPROVED** [review]
Looks good to me now.

**Pull request merged**